### PR TITLE
`kserver --socket=./my/socket`

### PR DESCRIPTION
**Pull request by h0nzZik**
Specify the socket for KServer

Example
```
./k-distribution/bin/kserver --socket ./my-socket &
wait 10
env KSERVER_SOCKET=(pwd)/my-socket ./k-distribution/bin/kompile --help
```

### CODE REVIEW

File: kernel/src/main/java/org/kframework/kserver/KServerOptions.java [hunk 1]
```diff
@@ -15,4 +15,8 @@
 
     @Parameter(names={"--port", "-p"}, description="The port to start the server on.")
     public int port = 2113;
+
+    @Parameter(names={"--socket"}, description="The socket to start the server on.")
```
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/kserver/KServerOptions.java]: this should say something like "The directory to put the unix domain socket in"

**Review by dwightguth — CHANGES_REQUESTED** [review]
looks good and potentially very valuable but can you please modify the pom so that mvn verify makes use of this functionality?

**Comment by h0nzZik** [comment]
@dwightguth Can you spot any obvious reason why is my integration (`pom.xml` and `spawn-kserver`) broken?

**Comment by dwightguth** [comment]
@h0nzZik this looks like the same bug that @ehildenb is trying to fix. If you just wait a couple days it should hopefully be resolved.

**Review by dwightguth — APPROVED** [review]

File: kernel/src/main/java/org/kframework/kserver/KServerOptions.java (hunk 1 above)
Comment (h0nzZik) [reply kernel/src/main/java/org/kframework/kserver/KServerOptions.java]: Done.

**Review by dwightguth — APPROVED** [review]

**Pull request merged**